test/unit/util/test_cuda_sparse_matrix_vector_ordering.cpp

```diff
@@ -0,0 +1,230 @@
+#include <gtest/gtest.h>
+
+#include <micm/util/cuda_sparse_matrix.hpp>
+#include <micm/util/sparse_matrix_vector_ordering.hpp>
+
+#include "test_sparse_matrix_policy.hpp"
+#include "cuda_matrix_utils.cuh"
+
+TEST(CudaSparseVectorMatrix, ZeroMatrix)
+{
+  auto matrix = testZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<2>>();
+
+  matrix.CopyToDevice();
+  auto param = matrix.AsDeviceParam();
+  micm::cuda::AddOneDriver(param);
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+
+  matrix.CopyToHost();
+
+  for(auto& elem : matrix.AsVector())
+  {
+    EXPECT_EQ(elem, 1.0);
+  }
+}
+
+TEST(CudaSparseVectorMatrix, CopyAssignmentConstZeroMatrix)
+{
+  const auto matrix = testConstZeroMatrix<micm::CudaSparseMatrix, micm::SparseMatrixVectorOrdering<3>>();
+  auto oneMatrix = matrix;
+  oneMatrix.CopyToDevice();
```

Comment:
why is a `CopyToDevice()` needed here? Should this be called on `matrix` prior to the copy assign?

Reply:
So my thoughts here were that we'd create the const matrix and copy the data to the device on the new matrix and test that the original matrix stays the same throughout the process.